evalml/preprocessing/data_splitters/time_series_split.py

```diff
@@ -60,6 +63,13 @@ def split(self, X, y=None, groups=None):
             split_kwargs = dict(X=y, groups=groups)
             max_index = y.shape[0]
         else:
+            if self.date_index:
+                X[self.date_index] = pd.to_datetime(X[self.date_index])
+                X = X.sort_values(by=self.date_index)
+                if not self._check_if_empty(y):
+                    y = y.reindex(index=X.index)
+                    y = y.reset_index(drop=True)
+                X = X.reset_index(drop=True)
```

Comment:
I wonder if we should be sorting the data in the data splitter. I think with the way this is implemented, the data passed to the pipelines would still not be sorted. 

Rather than fixing the user's data in this PR, I think it may be better to leave the sorting of the time index to a data check action?

Reply:
So initially I wasn't going to touch this but it seems that ARIMA was having issues post TimeSeriesSplit because of the `DelayedFeatureTransformer` I think. I forgot that as we removed `DateTimeFeaturizer` and `DelayedFeatureTransformer` from ARIMA pipelines it no longer became necessary, so I'll remove it thanks!